openapi_python_client/parser/properties/__init__.py

```diff
@@ -317,6 +317,11 @@ def build_model_property(
             optional_properties.append(prop)
         relative_imports.update(prop.get_imports(prefix=".."))
 
+    discriminator_mappings: Dict[str, Property] = {}
+    if data.discriminator is not None:
+        for k, v in (data.discriminator.mapping if data.discriminator else {}).items():
```

Comment:
Can we do this since the above line checks data.discriminator is not None?
```suggestion
        for k, v in data.discriminator.mapping.items():
```

Reply:
oh, good catch, thanks!

Reply:
> oh, good catch, thanks!

Well, hmm. I think I had that there because `data.discriminator` might be `{mapping: None}`.

Reply:
If `{"mapping": None}` then this will still return `None` since the check is on `data.discriminator` right?
May be worth a closer look.

Reply:
I have something like this in mind:
```python
from openapi_python_client.schema.discriminator import Discriminator
from openapi_python_client.schema import Schema

data = Schema(discriminator=Discriminator(propertyName="resourceType"))

>>> data.discriminator is None
False
>>> data.discriminator.mapping is None
True
>>> [(k,v) for k,v in data.discriminator.mapping.items()]
*** AttributeError: 'NoneType' object has no attribute 'items'
```

Reply:
Your intent makes sense.
To make your code match your intent I think you need to change the code from:
```
(data.discriminator.mapping if data.discriminator else {}).items():
```
to:
```
(data.discriminator.mapping if data.discriminator.mapping else {}).items():
```

alternatively you could replace the block with [EAFP](https://docs.python.org/3/glossary.html#term-EAFP)
```
try:
    for k, v in data.discriminator.mapping.items():
        discriminator_mappings[k] = Reference.from_ref(v)
except AttributeError:
    # Either data.discriminator or data.discriminator.mapping was not set
    pass
```

Reply:
You are right indeed, thanks! hopefully fixed this.